src/i18n-scanner.js

```diff
@@ -2,38 +2,61 @@ const fs = require('fs');
 const glob = require('glob');
 const Parser = require('i18next-scanner').Parser;
 
-const translationFunctionNames = ['i18next.t', 'props.t', 'this.props.t', 't'];
-const outputFilePath = './src/locales/en/common.json';
-
-const translationsSource = JSON.parse(fs.readFileSync(outputFilePath, 'utf8'));
-
 const parser = new Parser({
   keySeparator: '>',
   nsSeparator: '|',
 });
 
 
-const customHandler = function (key, options) {
-  const value = translationsSource[key] || key;
-  if (options.context) {
-    key += `_${options.context}`;
+function i18nScanner(params) {
+  const sourceJSON = fs.readFileSync(params.outputFilePath, 'utf8');
+  let translationsSource;
+  try {
+    translationsSource = JSON.parse(sourceJSON);
+  } catch (e) {
+    process.stderr.write(`i18nScanner: ${e}\n`);
+    return;
   }
-  parser.set(key, value);
-  if (options.count !== undefined) {
-    key = `${key}_plural`;
-    parser.set(key, translationsSource[key] || '');
+
+  const customHandler = function (key, options) {
+    const value = translationsSource[key] || key;
+    if (options.context) {
+      key += `_${options.context}`;
+    }
+    parser.set(key, value);
+    if (options.count !== undefined) {
+      key = `${key}_plural`;
+      parser.set(key, translationsSource[key] || '');
+    }
+  };
+
+  params.files.map(filePattern => glob.sync(filePattern, {}))
+    .reduce(((accumulator, files) => [...accumulator, ...files]), [])
+    .forEach((file) => {
+      const content = fs.readFileSync(file, 'utf-8');
+      parser.parseFuncFromString(content, { list: params.translationFunctionNames }, customHandler);
+    });
+
+  const translations = parser.get({ sort: true }).en.translation;
+  const count = Object.keys(translations).length;
+  const outputJSON = `${JSON.stringify(translations, null, 2)}\n`;
+  if (outputJSON !== sourceJSON) {
+    fs.writeFileSync(params.outputFilePath, outputJSON);
+    process.stdout.write(`i18nScanner: ${count} translation keys parsed and written to '${params.outputFilePath}'\n`);
   }
-};
+}
 
-const files = glob.sync('./src/**/*.js', {});
-const electronFiles = glob.sync('./app/src/**/*.js', {});
-[...files, ...electronFiles].forEach((file) => {
-  const content = fs.readFileSync(file, 'utf-8');
-  parser.parseFuncFromString(content, { list: translationFunctionNames }, customHandler);
-});
+class HelloWorldPlugin {
```

Comment:
Hello world plugin? I guess you've copied this from the tutorial sample. please rename to a more relevant name.